### DISCUSSION TITLE
How  to disable interaction testing for published storybook?

### DISCUSSION THREAD

**Discussion by lifan0127** [storybookjs/storybook#17949] [Help] [answered]
The recently added interaction testing feature has been great for development. But in some scenarios we may not want to auto-play all the interactions. Is there a way to disable interaction testing in setting?

**Comment by alexis-regnaud** [comment]
Any update on that one? Autoplay the interactions when we open a story looks weird (Especially when the Interactions tab is not the one focused)

**Reply by yannbf** [reply, marked as answer]
Hey there! There is currently no way to disable the play function to automatically run when visiting it. You can only disable the play function from automatically running [in docs mode](https://storybook.js.org/docs/web-components/api/doc-block-story#autoplay).

Regarding the interactions panel, you can actually make a story select an addon panel when visiting it by using the `options` parameter. You can select the interactions panel like so:
```ts
export const WithInteractions = {
  parameters: {
    options: { selectedPanel: 'storybook/interactions/panel' },
  }
}
```

**Reply by Kristian110** [reply]
Is there any change ?
Is it possible to stop autoplay, now ?
Or how to stop play function for not deployed version ?

**Reply by manandkumaar** [reply]
@yannbf I tried overriding **playFunction** in **beforeEach** in **preview.js** file, but with no luck. 
I am not sure if this is the right context to override. 

Is there any workaround to stop running play functions?.

**Reply by yannbf** [reply]
Hey everyone, there is currently no way to disable auto-playing a story, but you can check this workaround:
https://github.com/storybookjs/storybook/discussions/23616#discussioncomment-11339302

**Reply by rojasjandro89** [reply]
I'm sorry but this is bonkers. We have controls to literally control the execution of the test step by step (in both directions!). 

This renders the story useless from a presentational standpoint. Can't give this to a client and expect them to wait trough the tests every time they want to interact with the stories.  This and the fact that signals API don't get properly represented in the args table is forcing me to find other alternatives.
